### PR TITLE
[Enhancement] remove snapshot phase for olap table export

### LINKED ISSUE #16457: remove snapshot phase for olap table export
## Enhancement
The snapshot created by ExportPendingTask for olap table is not used by ExportExportingTask,
we can remove the snapshot phase.

### PR DESCRIPTION
**Pull request by xiaoyong-z**
Signed-off-by: xyz <a997647204@gmail.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #16457

## Problem Summary(Required) ：

The snapshot created by ExportPendingTask for olap table is not used by ExportExportingTask, in this pr, we remove the snapshot phase

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
- [ ] 2.5
- [ ] 2.4
- [ ] 2.3
- [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 7 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|0|2|00.00%|[612, 906]|
|:large_blue_circle:|com/starrocks/task/ExportPendingTask.java|0|5|00.00%|[94, 96, 97, 98, 99]|

**Comment by wyb** [comment]
I think we should have a good plan first and then optimize it. 
If be supports reading from snapshot, will there be no problem that the version cannot be read due to compaction? 
If the snapshot is really not needed, we should remove all snapshot-related codes, such as `snapshotPaths` variable and `releaseSnapshotPaths` function in `ExportJob`, `makeSnapshot` function in `ExportPendingTask`.

File: fe/fe-core/src/main/java/com/starrocks/task/ExportPendingTask.java [hunk 1]
```diff
@@ -91,11 +91,13 @@ protected void exec() {
             return;
         }
 
-        // make snapshots
-        Status snapshotStatus = makeSnapshots();
-        if (!snapshotStatus.ok()) {
-            job.cancelInternal(ExportFailMsg.CancelType.RUN_FAIL, snapshotStatus.getErrorMsg());
-            return;
+        if (!job.exportOlapTable()) {
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/task/ExportPendingTask.java:94]: Export is not applicable for non-olap table.

**Review by chaoyli — DISMISSED** [review]

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by xiaoyong-z** [comment]
run DCO

**Comment by wangruin** [comment]
run all

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 7 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|0|2|00.00%|[612, 906]|
|:large_blue_circle:|com/starrocks/task/ExportPendingTask.java|0|5|00.00%|[94, 96, 97, 98, 99]|

**Review by chaoyli — DISMISSED** [review]

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by xiaoyong-z** [comment]
run starrocks_be_unittest

**Comment by xiaoyong-z** [comment]
run starrocks_fe_unittest

**Comment by xiaoyong-z** [comment]
run starrocks_fe_unittest

**Comment by xiaoyong-z** [comment]
run starrocks_fe_unittest

**Comment by xiaoyong-z** [comment]
@mergify rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 7 (28.57%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/ExportPendingTask.java|1|5|20.00%|[96, 97, 98, 99]|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|1|2|50.00%|[612]|

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Pull request merged**